src/components/scenes/RequestScene.js

```diff
@@ -272,8 +286,9 @@ export class Request extends Component<Props, State> {
   }
 
   onExchangeAmountChanged = async (amounts: ExchangedFlipInputAmounts) => {
-    const { publicAddress, legacyAddress } = this.state
-    const edgeEncodeUri: EdgeEncodeUri = this.props.useLegacyAddress && legacyAddress ? { publicAddress, legacyAddress } : { publicAddress }
+    const { segwitAddress, publicAddress, legacyAddress } = this.state
+    const edgeEncodeUri: EdgeEncodeUri =
+      this.props.useLegacyAddress && legacyAddress ? { publicAddress, legacyAddress } : segwitAddress ? { segwitAddress, publicAddress } : { publicAddress }
```

Comment:
I looked at the types, and it seems like `EdgeEncodeIUri` does take a `segwitAddress` parameter, which seems wrong to me. I'm not in the mood to refactor this right now, so it can stay as-is.